### PR TITLE
C5G7 long tests

### PR DESCRIPTION
**Pull request by Naktakala**
## Highlights
- Changed workflow to run class01 on merges into the development branch
- Created a set of long tests for C5G7
- Corrected and updated test system documentation
- Corrected gitignore to ignore .vtu and .pvtu
- Add enhancements to the test system

### CODE REVIEW

File: test/modules/LinearBoltzmannSolvers/Transport_Keigen/c5g7/c5g7.lua [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+            angle_aggregation_type = "polar",
+            angle_aggregation_num_subsets = 1,
+            groupset_num_subsets = 1,
+        }
+    },
+    options =
+    {
+        boundary_conditions = { { name = "xmin", type = "reflecting"},
+                                { name = "ymin", type = "reflecting"} },
+        scattering_order = 1,
```
Comment (zachhardy) [inline test/modules/LinearBoltzmannSolvers/Transport_Keigen/c5g7/c5g7.lua:36]: Why use scattering order 1 when we only have isotropic scattering?

File: test/modules/LinearBoltzmannSolvers/Transport_Keigen/c5g7/materials/materials.lua [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    xs[tostring(m)] = chiPhysicsTransportXSCreate()
+end
+
+--chiPhysicsTransportXSSet(xs["0"],CHI_XSFILE,"materials/XS_water.cxs")
+--chiPhysicsTransportXSSet(xs["1"],CHI_XSFILE,"materials/XS_UO2.cxs")
+--chiPhysicsTransportXSSet(xs["2"],CHI_XSFILE,"materials/XS_4_3pMOX.cxs")
+--chiPhysicsTransportXSSet(xs["3"],CHI_XSFILE,"materials/XS_7pMOX.cxs")
+--chiPhysicsTransportXSSet(xs["4"],CHI_XSFILE,"materials/XS_8_7pMOX.cxs")
+--chiPhysicsTransportXSSet(xs["5"],CHI_XSFILE,"materials/XS_guide_tube.cxs")
+--chiPhysicsTransportXSSet(xs["6"],CHI_XSFILE,"materials/XS_fission_chamber.cxs")
```
Comment (zachhardy) [inline test/modules/LinearBoltzmannSolvers/Transport_Keigen/c5g7/materials/materials.lua]: Duplication. Remove.

File: test/modules/LinearBoltzmannSolvers/Transport_Keigen/c5g7/materials/materials.lua [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+chiPhysicsTransportXSSet(xs["4"],CHI_XSFILE,"materials/XS_4_3pMOX.cxs")
+chiPhysicsTransportXSSet(xs["5"],CHI_XSFILE,"materials/XS_8_7pMOX.cxs")
+chiPhysicsTransportXSSet(xs["6"],CHI_XSFILE,"materials/XS_fission_chamber.cxs")
+
+
+water_xs = chiPhysicsTransportXSGet(xs["0"])
+
+-- for k,v in pairs(water_xs) do
+--     print(k,v)
+-- end
```
Comment (zachhardy) [inline test/modules/LinearBoltzmannSolvers/Transport_Keigen/c5g7/materials/materials.lua]: Remove unless there is a compelling reason to keep.

File: test/modules/LinearBoltzmannSolvers/Transport_Keigen/c5g7/mesh/gmesh_coarse.lua [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+umesh = chiUnpartitionedMeshFromMshFormat("mesh/2D_c5g7_coarse.msh",true)
+
+--############################################### Create meshers
+chiSurfaceMesherCreate(SURFACEMESHER_PREDEFINED);
+chiVolumeMesherCreate(VOLUMEMESHER_UNPARTITIONED,umesh);
+
+--chiVolumeMesherSetKBAPartitioningPxPyPz(2,2,1)
+--chiVolumeMesherSetKBACutsX({27.72})
+--chiVolumeMesherSetKBACutsY({27.72})
+--chiVolumeMesherSetProperty(PARTITION_TYPE,KBA_STYLE_XYZ)
```
Comment (zachhardy) [inline test/modules/LinearBoltzmannSolvers/Transport_Keigen/c5g7/mesh/gmesh_coarse.lua]: Remove unless there is a compelling reason to keep.

File: test/modules/LinearBoltzmannSolvers/Transport_Keigen/c5g7/mesh/gmesh_refined.lua [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+umesh = chiUnpartitionedMeshFromMshFormat("mesh/2D_c5g7_refined.msh",true)
+
+--############################################### Create meshers
+chiSurfaceMesherCreate(SURFACEMESHER_PREDEFINED);
+chiVolumeMesherCreate(VOLUMEMESHER_UNPARTITIONED,umesh);
+
+--chiVolumeMesherSetKBAPartitioningPxPyPz(2,2,1)
+--chiVolumeMesherSetKBACutsX({27.72})
+--chiVolumeMesherSetKBACutsY({27.72})
+--chiVolumeMesherSetProperty(PARTITION_TYPE,KBA_STYLE_XYZ)
```
Comment (zachhardy) [inline test/modules/LinearBoltzmannSolvers/Transport_Keigen/c5g7/mesh/gmesh_refined.lua]: Remove unless there is a compelling reason to keep.

File: test/modules/LinearBoltzmannSolvers/Transport_Keigen/c5g7.tar.gz
Comment (zachhardy) [inline test/modules/LinearBoltzmannSolvers/Transport_Keigen/c5g7.tar.gz:1]: Why are we including a tar file here?

**Review by zachhardy — APPROVED** [review]
See general comments. Primarily semantic suggestions.

File: test/modules/LinearBoltzmannSolvers/Transport_Keigen/c5g7.tar.gz
Comment (Naktakala) [reply test/modules/LinearBoltzmannSolvers/Transport_Keigen/c5g7.tar.gz:1]: What is that doing there!? Deleted

File: test/modules/LinearBoltzmannSolvers/Transport_Keigen/c5g7/mesh/gmesh_refined.lua (hunk 5 above)
Comment (Naktakala) [reply test/modules/LinearBoltzmannSolvers/Transport_Keigen/c5g7/mesh/gmesh_refined.lua]: Deleted

File: test/modules/LinearBoltzmannSolvers/Transport_Keigen/c5g7/mesh/gmesh_coarse.lua (hunk 4 above)
Comment (Naktakala) [reply test/modules/LinearBoltzmannSolvers/Transport_Keigen/c5g7/mesh/gmesh_coarse.lua]: Deleted

File: test/modules/LinearBoltzmannSolvers/Transport_Keigen/c5g7/materials/materials.lua (hunk 3 above)
Comment (Naktakala) [reply test/modules/LinearBoltzmannSolvers/Transport_Keigen/c5g7/materials/materials.lua]: Deleted

File: test/modules/LinearBoltzmannSolvers/Transport_Keigen/c5g7/materials/materials.lua (hunk 2 above)
Comment (Naktakala) [reply test/modules/LinearBoltzmannSolvers/Transport_Keigen/c5g7/materials/materials.lua]: Deleted

File: test/modules/LinearBoltzmannSolvers/Transport_Keigen/c5g7.tar.gz
Comment (zachhardy) [reply test/modules/LinearBoltzmannSolvers/Transport_Keigen/c5g7.tar.gz:1]: I got my money on that being there from when you sent it to me a few days ago

File: test/modules/LinearBoltzmannSolvers/Transport_Keigen/c5g7/c5g7.lua (hunk 1 above)
Comment (Naktakala) [reply test/modules/LinearBoltzmannSolvers/Transport_Keigen/c5g7/c5g7.lua:36]: I am actually a fan of doing this. The scattering order does not ONLY control the scattering term's truncation. It also controls the fidelity to which the flux-moment expansion is computed. By setting L=1 you can get the current basically for free, i.e., `scalar_flux=phi_0` and `J=[phi_3, phi_1, phi_2]`. And view it in paraview if needed.

**Pull request merged**